File: MyLoftCoint/app/src/main/java/com/android/hootr/myloftcoint/di/module/AppModule.java
package com.android.hootr.myloftcoint.di.module;

import com.android.hootr.myloftcoint.data.api.Api;
import com.android.hootr.myloftcoint.data.db.Database;
import com.android.hootr.myloftcoint.data.db.model.CoinEntityMapper;
import com.android.hootr.myloftcoint.data.prefs.Prefs;
import com.android.hootr.myloftcoint.screens.main.rate.RatePresenter;
import com.android.hootr.myloftcoint.screens.main.rate.RatePresenterImpl;
import com.android.hootr.myloftcoint.screens.start.StartPresenter;
import com.android.hootr.myloftcoint.screens.start.StartPresenterImp;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module(includes = {PrefsModule.class, ApiModule.class, DatabaseModule.class})
public class AppModule {

    @Provides
    @Singleton
    public StartPresenter provideStartPresenter(Api api, Prefs prefs, Database dataBase) {
        CoinEntityMapper mapper = new CoinEntityMapper();
        return new StartPresenterImp(api, prefs, dataBase, mapper);
    }

    @Provides
    @Singleton
    public RatePresenter provideRatePresenter(Api api, Prefs prefs, Database dataBase) {
        CoinEntityMapper mapper = new CoinEntityMapper();
        return new RatePresenterImpl(api, prefs, dataBase, mapper);
    }
}

File: client1c/app/src/main/java/com/android/hootr/client1c/di/AppModule.kt
package com.android.hootr.client1c.di

import android.content.Context
import com.android.hootr.client1c.interactor.ImpMainInteractor
import com.android.hootr.client1c.interactor.MainInteractor
import com.android.hootr.client1c.prefs.SharedPrefsManager
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(private val context: Context) {

    @Provides
    @Singleton
    fun provideAppContext(): Context = context

    @Provides
    @Singleton
    fun provideMainInteractor(prefs: SharedPrefsManager): MainInteractor {
        return ImpMainInteractor(prefs)
    }

}